api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java

```diff
@@ -532,15 +532,10 @@ public Integer resolvePersonId(XCN xcn) throws HL7Exception {
 	@Override
 	@Transactional(readOnly = true)
 	public Integer resolveLocationId(PL pl) throws HL7Exception {
-		// TODO: Get rid of hack that allows first component to be an integer
-		// location.location_id
 		String pointOfCare = pl.getPointOfCare().getValue();
 		String facility = pl.getFacility().getUniversalID().getValue();
-		// HACK: try to treat the first component (which should be "Point of
-		// Care" as an internal openmrs location_id
 		try {
-			Integer locationId = Integer.valueOf(pointOfCare);
-			Location l = Context.getLocationService().getLocation(locationId);
```

Comment:
I don't think this is the change I expected, all you're doing is moving the logic from the service to the dao layer, I think the TODO is saying that the first component isn't supposed to be an id but doesn't suggest which component should be used instead.